x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformIndexerStats.java

```diff
@@ -73,8 +77,10 @@
         LENIENT_PARSER.declareLong(constructorArg(), NUM_INVOCATIONS);
         LENIENT_PARSER.declareLong(constructorArg(), INDEX_TIME_IN_MS);
         LENIENT_PARSER.declareLong(constructorArg(), SEARCH_TIME_IN_MS);
+        LENIENT_PARSER.declareLong(constructorArg(), PROCESSING_TIME_IN_MS);
```

Comment:
The new `long` values need to be `Long` since they won't exist in older stats files. This will throw an NPE if it tries to parse XContent where these new values don't exist.

Reply:
Thanks for pointing this out. I actually wonder why all variables are mandatory in the 1st place, especially as its supposed to be a `LENIENT_PARSER`, the exponential stats are optional.

Let me know if you have an idea why mandatory, if there is no good reason, I suggest to switch them all to optional.

Reply:
Will also make parsing more robust w.r.t. non existing fields (NPE)

Reply:
Switching them all to optional makes sense to me. The ctor should handling the `Long`. But I do think that the final stats object should just be unboxed `long` where `0L` is an unboxed `null`.

Reply:
I decided for a slightly different approach, to also avoid massive re-factorings, in a nutshell I use a "take default if Null" helper method. I am  surprised that I could not fine an existing method for this.